guides/micronaut-kafka/micronaut-kafka.adoc

```diff
@@ -0,0 +1,346 @@
+include::{commondir}/common-header.adoc[]
+
+In this guide, we will create two microservices that will use https://kafka.apache.org/[Kafka] to communicate with each other in an asynchronous and decoupled way.
+
+== What you will need
+
+To complete this guide, you will need the following:
+
+* Some time on your hands
+* A decent text editor or IDE
+* JDK 1.8 or greater installed with `JAVA_HOME` configured appropriately
+* https://www.docker.io/gettingstarted/#h_installation[Docker] and https://docs.docker.com/compose/install/[Docker Compose] installed if you will be running Kafka in Docker
+
+include::{commondir}/common-completesolution.adoc[]
+
+== Writing the application
+
+Let's describe the microservices you will build through the guide.
+
+* `books` - It returns a list of books. It uses a domain consisting of a book name and ISBN. It also publishes a message in Kafka every time a book is accessed.
+
+* `analytics` - It connects to Kafka to update the analytics for every book (a counter). It also exposes an endpoint to get the analytics.
+
+include::{commondir}/common-annotationprocessors.adoc[]
+
+=== Books Microservice
+
+Create the `books` microservice using the https://docs.micronaut.io/latest/guide/index.html#cli[Micronaut Command Line Interface] or with https://launch.micronaut.io[Micronaut Launch].
+
+[source,bash]
+----
+mn create-app --features=kafka,graalvm,testcontainers example.micronaut.books --build=@build@ --lang=@lang@
+----
+
+include::{commondir}/common-build-lang-arguments.adoc[]
+
+If you use Micronaut Launch, select Micronaut Application as application type and add the `kafka`, `graalvm`, and `testcontainers` features.
+
+The previous command creates a directory named `books` and a Micronaut application inside it with default package `example.micronaut`.
+
+In addition to the dependencies added by the `testcontainers` feature, we also need a test dependency for Kafka in Testcontainers, along with one for the http://www.awaitility.org/[Awaitility] library:
+
+:dependencies:
+dependency:kafka[groupId=org.testcontainers,scope=test]
+dependency:awaitility[groupId=org.awaitility,version=@awaitilityVersion@,scope=test]
+:dependencies:
```

Comment:
Test containers is mentioned here but there is no instructional content in the guide of where / how it is used

Reply:
Fixed